gen_mod_headers

```diff
@@ -242,7 +242,7 @@ if [[ -n "$prefix" ]]; then
 
             if [[ -e tools/build/Makefile ]]; then
 	        echo Building fixdep for target
-    	        make -C tools/build HOSTCC="$hostcc" $build_opts
+	        make -C tools/build HOSTCC="$hostcc" HOSTLD="$LD" $build_opts
```

Comment:
Since we are at it, shall we not pass it also in the 2 places were we cross compile above too?